### DISCUSSION TITLE
Cache Open MPI Build?

### DISCUSSION THREAD

**Discussion by mathomp4** [community/community#25976] [Actions]
<p>All,</p>
<p>I’m still at the neophyte stage of GitHub Actions, but I’m trying to learn. My current issue is I want to add CI for a project that uses MPI (and Fortran).</p>
<p>Now, being lazy I did try a <code>apt install openmpi-bin libopenmpi-dev</code> for my test, but it fails to work for me. Why? The Fortran I need to use (<code>gfortran-9</code>) was not the gfortran that the Open MPI package was built with. Thus <code>use mpi</code> fails.</p>
<p>Now, I have no problem building Open MPI (I’ve done it often), but it’s a long tedious build and as such I’d like to do it once rather than every time CI is spun up. I previously managed to hack my way into <a href="https://github.com/Goddard-Fortran-Ecosystem/pFUnit/blob/f61186ea982b56e9176caf3ce72a2a439dd5a848/tools/travis-install-mpi.sh" rel="nofollow noopener">getting TravisCI to do this</a> in <code>~/local</code> and then <a href="https://github.com/Goddard-Fortran-Ecosystem/pFUnit/blob/f61186ea982b56e9176caf3ce72a2a439dd5a848/.travis.yml#L54" rel="nofollow noopener">had Travis cache <code>~/local</code></a>.</p>
<p>Is something like this possible with GitHub Actions? That is, pay the price once to build Open MPI with <code>gfortran-9</code> and then use <code>actions/cache</code> to pull this pre-build Open MPI? I know I could roll my own Docker image to have this, but I want to see if I can learn caching instead.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/mathomp4">@mathomp4</a>,</p>
<p>Sure, you can use the <a href="https://github.com/actions/cache" rel="nofollow noopener"><strong>cache</strong> action</a> to do that in your workflows. The <strong>caching</strong> feature on <strong>GitHub Actions</strong> is designed to speed up time to get the outputs or downloaded dependencies that often reused in the workflow runs.</p>
<p>Generally, when you want to reuse dependencies that don’t change often between jobs or workflow runs, you can use <strong>caching</strong>.</p>
<p>To view more detailed information about <strong>caching</strong>, please see the official documentation below:<br>
<a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows</a></p>
